cmd/garm-cli/cmd/repository.go

```diff
@@ -55,17 +56,18 @@ var repoAddCmd = &cobra.Command{
 			return errNeedsInitError
 		}
 
-		newRepoReq := params.CreateRepoParams{
+		newRepoReq := apiClientRepos.NewCreateParams()
+		newRepoReq.Body = params.CreateRepoParams{
 			Owner:           repoOwner,
 			Name:            repoName,
 			WebhookSecret:   repoWebhookSecret,
 			CredentialsName: repoCreds,
 		}
-		repo, err := cli.CreateRepository(newRepoReq)
+		response, err := apiCli.Repositories.Create(newRepoReq)
```

Comment:
This is great as an example, but we either move everything to the generated client, or we leave everything as it is right now. When we do decide to migrate to the generated client, we need to also change the actual CLI. There is a `--debug` flag that gets passed to `resty` that prints the raw response. This is just an example, but a change to the actual CLI needs to be cohesive.

Reply:
To clarify, we can still have the generated client library live in this repo. But the actual CLI needs to be migrated later to use it. Until then, we can skip the change in this file.

Reply:
> This is great as an example, but we either move everything to the generated client, or we leave everything as it is right now. When we do decide to migrate to the generated client, we need to also change the actual CLI. There is a `--debug` flag that gets passed to `resty` that prints the raw response. This is just an example, but a change to the actual CLI needs to be cohesive.

https://github.com/cloudbase/garm/pull/119/files#diff-5014f3eeef4459bc7cc5b8f7cc8b57ac102b493d2aa20696e82feb1376ee1310R92

^ this does the trick to have the same output of the commands with `--debug` as we have with `resty`:
```go
if debug {
	os.Setenv("SWAGGER_DEBUG", "true")
}
```

this will make the swagger CLI autogenerated library to use debug mode on:
https://github.com/go-openapi/runtime/blob/1ade6d4fa0ddfcfdcad57761eed3588b40cf41ae/logger/logger.go#L11-L19

and the output of the `gram-cli --debug ...` is similar to what we have with `resty` debug enabled.

Reply:
> To clarify, we can still have the generated client library live in this repo. But the actual CLI needs to be migrated later to use it. Until then, we can skip the change in this file.

but this makes sense to me! I agree with it.